packages/file-upload/src/utils.tsx

```diff
@@ -69,24 +70,31 @@ export const getFileRejectionErrorMessage = (
 };
 
 export const getErrorSummary = (
-	rejections: FileError[],
+	rejections: RejectedFile[],
 	formattedMaxFileSize: string,
 	maxFiles: number | undefined
 ) => {
 	if (!rejections?.length) return;
 
-	const firstError = rejections[0];
+	const allErrorCodes = rejections
+		.reduce((acc, { errors }) => [...acc, ...errors], [] as FileError[])
+		.map(({ code }) => code)
+		.filter((value, index, self) => {
+			return self.indexOf(value) === index;
+		});
 
-	if (firstError.code === 'too-many-files') {
+	const firstError = allErrorCodes[0];
```

Comment:
```suggestion
	const firstErrorCode = uniqueErrorCodes[0];
```

Just being more explicit with the variable names